### ISSUE TITLE
Add global option for a rule to report "incomplete" instead of "failed"

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2186]
To improve configurability of axe-core, we'd like to have an option be available, either at the rule level or at the check level, to have its violations reported as incomplete instead.

**Comment by straker** [comment]
qa: create custom rule that overrides a current rule.id, then set `reviewOnFail` to true. Find a page that fails that rule and verify the rule instead returns as needs review:

```js
axe.configure({
  rules: [{
    id: 'region',
    reviewOnFail: true
  }]
});
```

**Comment by padmavemulapati** [comment]
For `regoin` rule fail testfile , `<body>
		<div>hello</div>
		foobar
	  </body>` 
initially it is in failure:
![image](https://user-images.githubusercontent.com/47409579/84616440-a1008400-aee9-11ea-85eb-8d72b9f554e8.png)
Then configured with : 
`axe.configure({
  rules: [{
    id: 'region',
    reviewOnFail: true
  }]
}); `
we can see the regoin rule into incomplete, which is expected to this cofiguration.
![image](https://user-images.githubusercontent.com/47409579/84616266-2899c300-aee9-11ea-926b-331d77af99f6.png)

**Comment by WilcoFiers** [comment]
This still needs documentation.

**Comment by straker** [comment]
Closed via https://github.com/dequelabs/axe-core/pull/2429
